util/fipstools/inject_hash_cpp/inject_hash.cpp

```diff
@@ -0,0 +1,61 @@
+#include <LIEF/LIEF.hpp>
+#include <iostream>
+#include <string>
+#include <cstring>
+
+static void print_usage() {
+    std::cout << "Usage: inject_hash_cpp -o output_file -in-object input_file [-apple]" << std::endl;
+}
+
+int main(int argc, char** argv) {
+    std::cout << "\n=== C++ inject_hash starting ===" << std::endl;
+
+    // Parse command line arguments
+    const char* input_file = nullptr;
+    const char* output_file = nullptr;
+    bool is_apple = false;
+
+    for (int i = 1; i < argc - 1; i++) {
+        if (strcmp(argv[i], "-in-object") == 0) {
+            input_file = argv[i + 1];
+            i++; // Skip next argument
+        }
+        else if (strcmp(argv[i], "-o") == 0) {
+            output_file = argv[i + 1];
+            i++; // Skip next argument
+        }
+        else if (strcmp(argv[i], "-apple") == 0) {
+            is_apple = true;
+        }
+    }
+
+    // Validate arguments
+    if (!input_file || !output_file) {
+        std::cerr << "❌ Missing required arguments" << std::endl;
```

Comment:
It is more helpful to say which ones are missing to help future customers/developers debug the issue.